File: C++/Estrutura repetitiva/fatorial.cpp
// Biblioteca(s)
#include <iostream>
#include <iomanip>

using namespace std;

int main()
{
    // Variável
    int fat, n, i;

    fat = 1;

    // Entrada de dados
    cout << "Digite um valor entre 0 a 15: ";
    cin >> n;

    // Estrutura repetitiva
    for (i=n; i > 0; i--) {
        // Algoritmo matemático
        fat *= i;
    }

    // Saída de dados
    cout << "O fatorial de " << n << "! = " << fat;

    return 0;
}

File: Java/Estrutura repetitiva/fatorial.java
// Biblioteca
import java.util.Scanner;

public class fatorial {

	public static void main(String[] args) {
		// Variável
		int fat, n, i;
		
		fat = 1;
		
		// Para facilitar a leitura
		Scanner sc = new Scanner(System.in);
		
		// Entrada de dados
		System.out.print("Digite um valor entre 0 a 15: ");
		n = sc.nextInt();
		
		// Estrutura repetitiva
		for (i = n; i > 0; i--) {
			// Algoritmo matemático
			fat *= i;
		}
		
	    // Saída de dados
		System.out.println("O fatorial de " + n + "! é " + fat);
		
		sc.close(); // Boa prática de programação
	}
}